presto-main/src/main/java/io/prestosql/security/ConnectorAccessControlForProcedures.java

```diff
@@ -0,0 +1,354 @@
+    {
+        checkArgument(context == null, "connectorSecurityContext should be null");
+        accessControl.checkCanGrantRoles(securityContext, roles, grantees, adminOption, grantor, catalogName);
+    }
+
+    @Override
+    public void checkCanRevokeRoles(ConnectorSecurityContext context, Set<String> roles, Set<PrestoPrincipal> grantees, boolean adminOption, Optional<PrestoPrincipal> grantor, String catalogName)
+    {
+        checkArgument(context == null, "connectorSecurityContext should be null");
+        accessControl.checkCanRevokeRoles(securityContext, roles, grantees, adminOption, grantor, catalogName);
+    }
+
+    @Override
+    public void checkCanSetRole(ConnectorSecurityContext context, String role, String catalogName)
+    {
+        checkArgument(context == null, "connectorSecurityContext should be null");
+        accessControl.checkCanSetRole(securityContext, role, catalogName);
+    }
+
+    @Override
+    public void checkCanShowRoleAuthorizationDescriptors(ConnectorSecurityContext context, String catalogName)
+    {
+        checkArgument(context == null, "connectorSecurityContext should be null");
+        accessControl.checkCanShowRoleAuthorizationDescriptors(securityContext, catalogName);
+    }
+
+    @Override
+    public void checkCanShowRoles(ConnectorSecurityContext context, String catalogName)
+    {
+        checkArgument(context == null, "connectorSecurityContext should be null");
+        accessControl.checkCanShowRoles(securityContext, catalogName);
+    }
+
+    @Override
+    public void checkCanShowCurrentRoles(ConnectorSecurityContext context, String catalogName)
+    {
+        checkArgument(context == null, "connectorSecurityContext should be null");
+        accessControl.checkCanShowCurrentRoles(securityContext, catalogName);
+    }
+
+    @Override
+    public void checkCanShowRoleGrants(ConnectorSecurityContext context, String catalogName)
+    {
+        checkArgument(context == null, "connectorSecurityContext should be null");
+        accessControl.checkCanShowRoleGrants(securityContext, catalogName);
+    }
+
+    @Override
+    public void checkCanExecuteProcedure(ConnectorSecurityContext context, SchemaRoutineName procedure)
+    {
+        checkArgument(context == null, "connectorSecurityContext should be null");
+        accessControl.checkCanExecuteProcedure(securityContext, new QualifiedObjectName(catalogName, procedure.getSchemaName(), procedure.getRoutineName()));
+    }
+
+    @Override
+    public Optional<ViewExpression> getRowFilter(ConnectorSecurityContext context, SchemaTableName tableName)
+    {
+        checkArgument(context == null, "connectorSecurityContext should be null");
+        return Optional.ofNullable(getFirst(accessControl.getRowFilters(securityContext, getQualifiedObjectName(tableName)), null));
```

Comment:
We should probably throw `IllegalStateException` here if there are multiple filters, since picking the first one would be discarding the other row filters. Ideally, the engine should probably analyze and combine them into a single expression, but we don't need to worry about this now.